src/tfs/tlog.c

```diff
@@ -924,6 +936,15 @@ log log_create(heap h, filesystem fs, boolean initialize, status_handler sh)
         log_read(tl, sh);
     }
     return tl;
+#ifndef TLOG_READ_ONLY
+  fail_ext_new:
+    filesystem_free_storage(fs, irangel(fs->next_extend_log_offset, filesystem_log_blocks(fs)));
+  fail_alloc_extend_log:
+    filesystem_free_storage(fs, irangel(fs->next_new_log_offset, filesystem_log_blocks(fs)));
+  fail:
+#endif
+    apply(sh, timm("result", "failed to allocate storage"));
+    return INVALID_ADDRESS;
```

Comment:
It seems that the `#endif` above should go below `return INVALID_ADDRESS`, since none of these new instructions can be executed when TLOG_READ_ONLY is defined

Reply:
Thanks - fixed